Assets/HoloToolkit/Input/Scripts/Utilities/AttachToController.cs

```diff
@@ -42,90 +40,56 @@ public class AttachToController : MonoBehaviour
 
         public bool IsAttached { get; private set; }
 
-        private Transform elementTransform;
-        public Transform ElementTransform { get; private set; }
-
-        protected MotionControllerInfo controller;
-
         protected virtual void OnAttachToController() { }
         protected virtual void OnDetachFromController() { }
 
-#if UNITY_WSA && UNITY_2017_2_OR_NEWER
-        protected virtual void OnEnable()
+        public override void OnEnable()
         {
             SetChildrenActive(false);
-
-            // Look if the controller has loaded.
-            if (MotionControllerVisualizer.Instance.TryGetControllerModel(handedness, out controller))
-            {
-                AttachElementToController(controller);
-            }
-
-            MotionControllerVisualizer.Instance.OnControllerModelLoaded += AttachElementToController;
-            MotionControllerVisualizer.Instance.OnControllerModelUnloaded += DetachElementFromController;
+            base.OnEnable();
         }
 
-        protected virtual void OnDisable()
+        protected override void AddControllerTransform(MotionControllerInfo newController)
         {
-            if (MotionControllerVisualizer.IsInitialized)
-            {
-                MotionControllerVisualizer.Instance.OnControllerModelLoaded -= AttachElementToController;
-                MotionControllerVisualizer.Instance.OnControllerModelUnloaded -= DetachElementFromController;
-            }
-        }
-
-        protected virtual void OnDestroy()
-        {
-            if (MotionControllerVisualizer.IsInitialized)
+#if UNITY_WSA && UNITY_2017_2_OR_NEWER
+            if (!IsAttached)
```

Comment:
We should still check handedness here, like it used to.

https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/1453/files#diff-b901af09abe8d8ffcedd390d233ef120L86

Reply:
Yep. Missed a conditional. I'll update. Sorry, pulling the if def changes and trying to rectify it with my existing changes made for some confusion.

Reply:
I was originally wanting to have the base methods handle handedness and *AttachToController* handle the *IsAttached* logic. Hence the missed conditional - Its not that simple. :)